futures.js

```diff
@@ -0,0 +1,32 @@
+'use strict';
+
+const fs            = require('fs');
+
+const Future        = require('fluture');
+
+const exit0         = require('./common/exit0');
+const exit1         = require('./common/exit1');
+const join          = require('./common/join');
+const S             = require('./common/sanctuary');
+
+
+//    readFile :: String -> Future Error String
+const readFile =
+S.compose(Future.node, S.curry3(fs.readFile, S.__, {encoding: 'utf8'}));
+
+//    concatFiles :: (String -> String) -> Future Error String
+const concatFiles = path =>
+  S.pipe([path,                                   // :: String
+          readFile,                               // :: Future Error String
+          S.map(S.lines),                         // :: Future Error (Array String)
+          S.map(S.map(path)),                     // :: Future Error (Array String)
+          S.chain(S.traverse(Future, readFile)),  // :: Future Error (Array String)
```

Comment:
Reading files in parallel would be very nice indeed! Which version do you favour, and why? I'm happy to follow your recommendation.